### PR TITLE
[branch-2.9][cherry-pick] Close TransactionBuffer when MessageDeduplication#checkStatus failed

### PR DESCRIPTION
**Pull request by tjiuming**
### Motivation

Cherry-pick https://github.com/apache/pulsar/pull/19157 into branch-2.9

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Comment by codelipenghui** [comment]
@tjiuming Could you please help check the failed test? It looks like introduced by this PR.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 1]
```diff
@@ -1325,6 +1325,12 @@ private void checkOwnershipAndCreatePersistentTopic(final String topic, boolean
                 });
     }
 
+    @VisibleForTesting
+    public void createPersistentTopic0(final String topic, boolean createIfMissing,
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:1329]: why don't use `BrokerService#getTopic`?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 2]
```diff
@@ -1400,6 +1406,12 @@ public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
                                             "Replication or dedup check failed."
                                                     + " Removing topic from topics list {}, {}",
                                             topic, ex);
+                                    persistentTopic.getTransactionBuffer()
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:1409]: Should we add this similar logic to line `1395`?

**Review by mattisonchao — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java (hunk 2 above)
Comment (tjiuming) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:1409]: Great catch!
it makes sense, in line `1395`, it seems like checking create topic timeout, it's also need to close transaction buffer.
I'll create another PR to add the logic

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java (hunk 1 above)
Comment (tjiuming) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:1329]: yes we can use the method, but when I wrote the test, I chose this method

**Review by gaoran10 — APPROVED** [review]

**Pull request merged**